crates/biome_analyze/CONTRIBUTING.md

```diff
@@ -103,54 +105,154 @@ New rules **must** be placed inside the `nursery` group. This group is meant as
 
    ```shell
+   # Example: Create a new JS lint rule
    just new-js-lintrule useMyRuleName
+
+   # Or, to create a new lint/assist rule for JSON/CSS/GraphQL/..:
+   # $ just new-css-lintrule useMyRuleName
+   # $ just new-graphql-lintrule useMyRuleName
+   # $ just new-js-assistrule useMyRuleName
+   # $ just new-json-assistrule useMyRuleName
    ```
-   The script will generate a bunch of files for the _JavaScript_ language, inside the `biome_js_analyze` crate.
-   Among the other files, you'll find a file called `use_my_new_rule_name.rs` inside the `biome_js_analyze/lib/src/lint/nursery` folder. You'll implement your rule in this file.
+   The script `just new-js-lintrule` script will generate a bunch of files for the _JavaScript_ language inside the `biome_js_analyze` crate.
+   Among the other files, you'll find a file called `use_my_rule_name.rs` inside the `biome_js_analyze/lib/src/lint/nursery` folder. You'll implement your rule in this file.
 
-   If you want to create a _CSS_ lint rule, run this script. It will generate a new lint rule in `biome_css_analyze`
+2. Let's have a look at the generated code in  `use_my_rule_name.rs`:
 
-   ```shell
-   just new-css-lintrule useMyRuleName
+
+   ```rust
+   ...
+   impl Rule for UseMyRuleName {
+       type Query = Ast<JsIdentifierBinding>;
+       type State = ();
+       type Signals = Option<Self::State>;
+       type Options = ();
+
+       fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+           ...
+       }
+
+       fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+           ...
+       }
+   }
    ```
 
-1. The `Ast` query type allows you to query the CST of a program.
-1. The `State` type doesn't have to be used, so it can be considered optional. However, it has to be defined as `type State = ()`.
-1. Implement the `run` function:
+   We can observe a few details from this snippet:
+
+   - The **`Options`** type can be used to define [additional options](#rule-options) for your rule:
+
+     ```rust
+     type Options = ();
+     ```
+
+     Use `()` if your rule does not have additional options.
+
+   - The **`Query`** type defines the entities for which your your rule's `UseMyRuleName::run` function will be invoked:
+     ```rust
+     type Query = Ast<JsIdentifierBinding>;
+     ```
+
+     _→ The `Ast<>` query type for example allows you to query the AST/CST of a program._
+
+     _→ For more advanced use cases it is also possible to define [custom query types](#custom-syntax-tree-visitors)._
+
+   - The **`run`** function will be invoked for each match of `Query`.
+     It should return either `Some` to report a diagnostic for that match, or `None`.
+
+     _→ It is also possible to report multiple diagnostics for a single `Query` match._
+     _See [Multiple Signals](#multiple-signals) for instructions._
```

Comment:
I would avoid using the term "multiple diagnostics" because it isn't true. The analyzer already returns multiple diagnostics for each query match. Having multiple *signals* for a query match isn't about diagnostics per se, but it's more about code actions and "awareness" of the rule.

Reply:
I still need to find a good wording to explain when to use multiple signals, so it's definitely something we need to clarify eventually

Reply:
Maybe it would be a good idea to include a link to the documentation (or code) for the `Rule` trait, because (after reading it just now....), the documentation seems concise and good, covering the different aspects of `query matches`, `signals`, `instances`, `actions`, `diagnostics` and their relationship.

Reply:
The `Rule` trait has many functions, but many of them were made for some edge cases, hence they are less used. I wasn't around when `suppressed_nodes` was made, but was created to stop the rule from emitting signals inside some recursed code. I think we use it inside the template literals code. 

Happy to improve the docs as much as we can!